pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -625,10 +627,24 @@ func ConsolidateObjects(env api.Environment, cr *api.KieApp) api.Environment {
 	for index := range env.Servers {
 		serverSet, _ := GetServerSet(cr, index)
 		env.Servers[index] = ConstructObject(env.Servers[index], serverSet.KieAppObject)
+		setCRBuildConfigEnvsToSpec(cr, &env)
```

Comment:
since, at this point in code, we already have the server index... should we instead just pass in the exact server object?

Reply:
yeah, remove the extra func, it is cleaner now.